Bitwise AND between `BigInt` that come from `PrimeField`

Hi everyone, 

I wan to implement the following function considering the last contribution to bitwise operations between `BigInt` (https://github.com/arkworks-rs/algebra/pull/713):

```rust
use ark_ff::fields::PrimeField;

fn compute_and<F: PrimeField>(a: F, b: F) -> F {
    let a_bigint = a.into_bigint();
    let b_bigint = b.into_bigint();
    let and_bigint =  a_bigint & b_bigint;
    let and_field = match F::from_bigint(and_bigint) {
        Some(value) => value,
        None => panic!("Something bad happened."),
    };

    and_field
}
```

This code is incorrect, and the output shows the following error which I can understand:

```
error[E0308]: mismatched types
  --> src/main.rs:7:42
   |
6  |     let and_bigint =  a_bigint & b_bigint;
   |                       ------------------- here the type of `and_bigint` is inferred to be `<<F as PrimeField>::BigInt as BitAnd>::Output`
7  |     let and_field = match F::from_bigint(and_bigint) {
   |                           -------------- ^^^^^^^^^^ expected `ark_ff::PrimeField::BigInt`, found `std::ops::BitAnd::Output`
   |                           |
   |                           arguments to this function are incorrect
   |
   = note: expected associated type `<F as PrimeField>::BigInt`
              found associated type `<<F as PrimeField>::BigInt as BitAnd>::Output`
   = note: an associated type was expected, but a different one was found
note: associated function defined here
  --> /home/hdvanegasm/.cargo/git/checkouts/algebra-7e23afa68841b66e/b7b988f/ff/src/fields/prime.rs:55:8
   |
55 |     fn from_bigint(repr: Self::BigInt) -> Option<Self>;
   |        ^^^^^^^^^^^

For more information about this error, try `rustc --explain E0308`.
error: could not compile `ark` (bin "ark") due to previous error
```

Is it possible to implement this function for arbitrary `PrimeField` elements with the current state of the library, or do we need to extend the functionality by submitting a PR?

--- a/ff/src/biginteger/mod.rs
+++ b/ff/src/biginteger/mod.rs
@@ -799,15 +799,15 @@ pub trait BigInteger:
     + Into<BigUint>
     + BitXorAssign<Self>
     + for<'a> BitXorAssign<&'a Self>
-    + BitXor<Self>
+    + BitXor<Self, Output = Self>
     + for<'a> BitXor<&'a Self, Output = Self>
     + BitAndAssign<Self>
     + for<'a> BitAndAssign<&'a Self>
-    + BitAnd<Self>
+    + BitAnd<Self, Output = Self>
     + for<'a> BitAnd<&'a Self, Output = Self>
     + BitOrAssign<Self>
     + for<'a> BitOrAssign<&'a Self>
-    + BitOr<Self>
+    + BitOr<Self, Output = Self>
     + for<'a> BitOr<&'a Self, Output = Self>
 {
     /// Number of 64-bit limbs representing `Self`.

--- a/ff/src/biginteger/tests.rs
+++ b/ff/src/biginteger/tests.rs
@@ -1,5 +1,4 @@
-use crate::{biginteger::BigInteger, BigInt, UniformRand};
-use ark_std::rand::Rng;
+use crate::{biginteger::BigInteger, UniformRand};
 use num_bigint::BigUint;
 
 // Test elementary math operations for BigInteger.
@@ -58,43 +57,43 @@ fn biginteger_bitwise_ops_test<B: BigInteger>() {
 
     // Test XOR
     // a xor a = 0
-    let a: BigInt<4> = UniformRand::rand(&mut rng);
-    assert_eq!(a ^ &a, BigInt::from(0_u64));
+    let a = B::rand(&mut rng);
+    assert_eq!(a ^ &a, B::from(0_u64));
 
     // Testing a xor b xor b
-    let a: BigInt<4> = UniformRand::rand(&mut rng);
-    let b: BigInt<4> = UniformRand::rand(&mut rng);
+    let a = B::rand(&mut rng);
+    let b = B::rand(&mut rng);
     let xor_ab = a ^ b;
     assert_eq!(xor_ab ^ b, a);
 
     // Test OR
     // a or a = a
-    let a = rng.gen::<BigInt<4>>();
+    let a = B::rand(&mut rng);
     assert_eq!(a | &a, a);
 
     // Testing a or b or b
-    let a = rng.gen::<BigInt<4>>();
-    let b = rng.gen::<BigInt<4>>();
+    let a = B::rand(&mut rng);
+    let b = B::rand(&mut rng);
     let or_ab = a | b;
     assert_eq!(or_ab | &b, a | b);
 
     // Test AND
     // a and a = a
-    let a = rng.gen::<BigInt<4>>();
+    let a = B::rand(&mut rng);
     assert_eq!(a & (&a), a);
 
     // Testing a and a and b.
-    let a = rng.gen::<BigInt<4>>();
-    let b = rng.gen::<BigInt<4>>();
+    let a = B::rand(&mut rng);
+    let b = B::rand(&mut rng);
     let b_clone = b.clone();
     let and_ab = a & b;
     assert_eq!(and_ab & b_clone, a & b);
 
     // Testing De Morgan's law
-    let a = rng.gen::<BigInt<4>>();
-    let b = rng.gen::<BigInt<4>>();
-    let de_morgan_lhs = !(a | b);
-    let de_morgan_rhs = (!a) & (!b);
+    let a = 0x1234567890abcdef_u64;
+    let b = 0xfedcba0987654321_u64;
+    let de_morgan_lhs = B::from(!(a | b));
+    let de_morgan_rhs = B::from(!a) & B::from(!b);
     assert_eq!(de_morgan_lhs, de_morgan_rhs);
 }
 

